### PR TITLE
[SPARK-41782][TESTS] Regenerate benchmark results

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

 This aims to regenerate benchmark results on `master` branch as a part of preparing Apache Spark 3.4.0 release to identify potential regressions during QA period.

- Java 8: https://github.com/dongjoon-hyun/spark/actions/runs/3803071014
- Java 11: https://github.com/dongjoon-hyun/spark/actions/runs/3803075549
- Java 17: https://github.com/dongjoon-hyun/spark/actions/runs/3803077606

In addition,
- `MetadataStructBenchmark` benchmark is added by SPARK-37980, but the results are added here.

### Why are the changes needed?

This are reference values.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review.

### CODE REVIEW

File: sql/core/benchmarks/JoinBenchmark-jdk11-results.txt [hunk 1]
```diff
@@ -2,81 +2,81 @@
...
-OpenJDK 64-Bit Server VM 11.0.14+9-LTS on Linux 5.13.0-1021-azure
-Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
+OpenJDK 64-Bit Server VM 11.0.17+8 on Linux 5.15.0-1023-azure
+Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
 Join w 2 ints:                            Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Join w 2 ints wholestage off                     320974         321015          57          0.1       15305.3       1.0X
-Join w 2 ints wholestage on                      237636         238622         567          0.1       11331.4       1.4X
+Join w 2 ints wholestage off                     206309         209960        2865          0.1        9837.6       1.0X
+Join w 2 ints wholestage on                      260860         265467        1202          0.1       12438.8       0.8X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/JoinBenchmark-jdk11-results.txt:24]: This looks like a regression.

File: sql/core/benchmarks/JoinBenchmark-jdk17-results.txt [hunk 2]
```diff
@@ -2,81 +2,81 @@
...
-OpenJDK 64-Bit Server VM 17.0.2+8-LTS on Linux 5.13.0-1021-azure
-Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
+OpenJDK 64-Bit Server VM 17.0.5+8 on Linux 5.15.0-1023-azure
+Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
 Join w 2 ints:                            Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Join w 2 ints wholestage off                     358176         358191          22          0.1       17079.1       1.0X
-Join w 2 ints wholestage on                      207044         207239         239          0.1        9872.6       1.7X
+Join w 2 ints wholestage off                     169271         170528        1778          0.1        8071.5       1.0X
+Join w 2 ints wholestage on                      162252         164248         NaN          0.1        7736.8       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/JoinBenchmark-jdk17-results.txt:24]: ditto.

File: sql/core/benchmarks/JoinBenchmark-results.txt [hunk 3]
```diff
@@ -2,81 +2,81 @@
...
-OpenJDK 64-Bit Server VM 1.8.0_322-b06 on Linux 5.13.0-1021-azure
-Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz
+OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
+Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
 Join w 2 ints:                            Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Join w 2 ints wholestage off                     426979         427823        1194          0.0       20359.9       1.0X
-Join w 2 ints wholestage on                      208958         209733         509          0.1        9963.9       2.0X
+Join w 2 ints wholestage off                     183695         184609        1292          0.1        8759.3       1.0X
+Join w 2 ints wholestage on                      180862         181268         311          0.1        8624.2       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/JoinBenchmark-results.txt:24]: ditto.

File: sql/core/benchmarks/MetadataStructBenchmark-jdk11-results.txt [hunk 4]
```diff
@@ -0,0 +1,40 @@
+================================================================================================
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/MetadataStructBenchmark-jdk11-results.txt:1]: This seems to be missed at SPARK-37980.

File: sql/hive/benchmarks/OrcReadBenchmark-results.txt [hunk 5]
```diff
@@ -2,221 +2,221 @@
...
+OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
+Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
 Single Struct Column Scan with 600 Fields:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 -------------------------------------------------------------------------------------------------------------------------
-Hive built-in ORC                                  27779          28882        1559          0.0       26492.5       1.0X
-Native ORC MR                                      29329          29488         226          0.0       27970.1       0.9X
-Native ORC Vectorized                              30460          30772         441          0.0       29049.4       0.9X
+Hive built-in ORC                                  32291          36141         NaN          0.0       30795.2       1.0X
+Native ORC MR                                      94939          95045         149          0.0       90541.2       0.3X
+Native ORC Vectorized                              93062          93335         386          0.0       88750.4       0.3X
```
Comment (dongjoon-hyun) [inline sql/hive/benchmarks/OrcReadBenchmark-results.txt:191]: I'll take a look at this, `Single Struct Column Scan with 600 Fields`.

**Comment by dongjoon-hyun** [comment]
Could you review this PR, @HyukjinKwon ?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Should be good to go since we don't test this in CI

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon . Yes, right. The yarn ut failure is a flaky one. I'll merge this to proceed more.

**Comment by LuciferYang** [comment]
late LGTM

**Comment by dongjoon-hyun** [comment]
Hi, FYI, @HyukjinKwon and @LuciferYang .
I found the root cause of regression. Here is my comment about that Spark commit.
- https://github.com/apache/spark/pull/37280#pullrequestreview-1234276498

**Comment by dongjoon-hyun** [comment]
I created a PR to fix a perf regression.
- https://github.com/apache/spark/pull/39362

**Pull request closed without merging**